config/application.rb

```diff
@@ -59,8 +59,15 @@ class Application < Rails::Application
     # MySQL bug where `rake db:reset` causes ActiveRecord to be loaded
     # before initializers and causes configuration not to be respected.
     config.load_defaults 6.0
+    # config.load_defaults 6.1
+
     config.active_record.belongs_to_required_by_default = false
     config.active_record.include_root_in_json = true
+
+    # Support for inversing belongs_to -> has_many Active Record associations.
+    # Overriding Rails 6.1 default, because it causes various issues
+    config.active_record.has_many_inversing = false
```

Comment:
The issues I found are commented in the places they occur, search for `has_many_inversing`.

Also, it seems that there are some open issues related to this setting:
https://github.com/rails/rails/issues/50258
https://github.com/rails/rails/issues/47559

and maybe more. So I think the safest way would be to keep it permanently, until a deeper review of this is performed.

Reply:
So without `has_many_inversing` what is the behavior? Yeah, it seems we better keep this forever. No need to change it. We can set `has_one` or `has_many` whenever needed but doing it automatically appears to be harmful as we can't override at some place but keep in other places.

I would suggest adding a comment about it.

Reply:
```suggestion
    # likely need to keep it forever as we can't override it for individual use cases
    # also the feature has outstanding bugs: rails/rails#47559 rails/rails#50258
    config.active_record.has_many_inversing = false
```